docs/quickstart.rst

```diff
@@ -61,198 +199,276 @@ instance had already existing in the destination directory, this command
+the nodes to connect to one another. For instance, pg_autoctl added the
+following lines to node A:
 
-All that differs here is we’re running it on another port and pointing
-at another data directory. It discovers from the monitor that a primary
-exists, and then switches its own state to be a hot standby and begins
-streaming WAL contents from the primary.
+.. code-block:: ini
+
+   # automatically added to node A
+
+   host "appdb" "ha-admin" ha-demo-a.internal.cloudapp.net trust
+   host all "pgautofailover_monitor" ha-demo-monitor.internal.cloudapp.net trust
+   host replication "pgautofailover_replicator" ha-demo-b.internal.cloudapp.net trust
+   host "appdb" "pgautofailover_replicator" ha-demo-b.internal.cloudapp.net trust
+   host replication "pgautofailover_replicator" ha-demo-a.internal.cloudapp.net trust
+   host "appdb" "pgautofailover_replicator" ha-demo-a.internal.cloudapp.net trust
+
+For ``pg_hba.conf`` on the monitor node pg_autoctl inspects the local network
+and makes its best guess about the subnet to allow. In our case it guessed
+correctly:
+
+.. code-block:: ini
+
+   # automatically added to the monitor
+
+   host "pg_auto_failover" "autoctl_node" 10.0.1.0/24 trust
+
+If worker nodes have more ad-hoc addresses and are not in the same subnet, it's
+better to disable pg_autoctl's automatic modification of pg_hba using the
+``--skip-pg-hba`` command line option during creation. You will then need to
+edit the hba file by hand. Another reason for manual edits would be to use
+special authentication methods.
 
 Watch the replication
 ---------------------
 
 First let’s verify that the monitor knows about our nodes, and see what
 states it has assigned them:
 
-.. code-block:: text
+.. code-block::
 
-   pg_autoctl show state --pgdata ./monitor
-        Name |   Port | Group |  Node |     Current State |    Assigned State
-   ----------+--------+-------+-------+-------------------+------------------
-   127.0.0.1 |   6010 |     0 |     1 |           primary |           primary
-   127.0.0.1 |   6011 |     0 |     2 |         secondary |         secondary
+   ssh -l ha-admin `vm_ip monitor` pg_autoctl show state --pgdata monitor
 
-This looks good. We can add data to the primary, and watch it get
-reflected in the secondary.
+                              Name |   Port | Group |  Node |     Current State |    Assigned State
+   --------------------------------+--------+-------+-------+-------------------+------------------
+   ha-demo-a.internal.cloudapp.net |   5432 |     0 |     1 |           primary |           primary
+   ha-demo-b.internal.cloudapp.net |   5432 |     0 |     2 |         secondary |         secondary
+
+
+This looks good. We can add data to the primary, and later see it appear in the
+secondary. We'll connect to the database from inside our "app" virtual machine,
+using a connection string obtained from the monitor. First we'll get the
+connection string and store it in a local environment variable:
```

Comment:
Maybe we could show the following output to show what we're dealing with, and then set the APP_DB_URI your way so that we can re-use the connection string?

```
~ ssh -l ha-admin `vm_ip monitor` -- pg_autoctl show uri --pgdata monitor
      Type |    Name | Connection String
-----------+---------+-------------------------------
   monitor | monitor | postgres://autoctl_node@ha-demo-monitor.internal.cloudapp.net:5432/pg_auto_failover?sslmode=require
 formation | default | postgres://ha-demo-a.internal.cloudapp.net:5432,ha-demo-b.internal.cloudapp.net:5432/appdb?target_session_attrs=read-write&sslmode=require
```